src/webapp/components/multiple-selector/useMultipleSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { useBooleanState } from "$/webapp/utils/use-boolean";
+import { MultipleSelectorProps } from "$/webapp/components/multiple-selector/MultipleSelector";
+import i18n from "$/utils/i18n";
+import _ from "$/domain/entities/generic/Collection";
+
+export function useMultipleSelector(props: MultipleSelectorProps) {
+    const {
+        items,
+        values,
+        onChange,
+        allOption,
+        customMenu,
+        type = "item",
+        pluralType = "items",
+    } = props;
+
+    const [menuIsOpen, { enable: openMenu, disable: closeMenu }] = useBooleanState(false);
+    const [tooltipIsOpen, { enable: openTooltip, disable: closeTooltip }] = useBooleanState(false);
+    const [filterText, setFilterText] = React.useState("");
+
+    const notifyChange = React.useCallback(
+        (event: React.ChangeEvent<{ value: unknown }>) => {
+            onChange(
+                (event.target.value as string[]).filter(
+                    s => s !== "multiple-selector-void" && s !== undefined && s !== null
+                )
+            );
+        },
+        [onChange]
+    );
+
+    const filter = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setFilterText(event.target.value);
+    }, []);
+
+    const isAllSelected = React.useMemo(
+        () => (allOption ? values.includes(allOption.value) : false),
+        [allOption, values]
+    );
+
+    const selected = React.useMemo(() => {
+        if (isAllSelected && allOption) return [allOption.value];
+        else return _(values).isEmpty() ? ["multiple-selector-void"] : values;
+    }, [allOption, isAllSelected, values]);
+
+    const helperText = React.useMemo(() => {
+        if (allOption && isAllSelected)
+            return `${items.length} ${items.length > 1 ? pluralType : type} ` + i18n.t("selected");
```

Comment:
Languages have different ways of saying things than 'N selected', so better to use a i18n string with interpolation.